flink-libraries/flink-table/src/main/scala/org/apache/flink/table/expressions/stringExpressions.scala

```diff
@@ -358,6 +358,29 @@ case class Rpad(text: Expression, len: Expression, pad: Expression)
   }
 }
 
+case class IsNumeric(child: Expression) extends UnaryExpression with InputTypeSpec {
+
+  override private[flink] def expectedTypes: Seq[TypeInformation[_]] = Seq(STRING_TYPE_INFO)
+
+  override private[flink] def resultType: TypeInformation[_] = INT_TYPE_INFO
+
+  override private[flink] def validateInput(): ValidationResult = {
+    if (child.resultType == STRING_TYPE_INFO) {
```

Comment:
Wasn't sure if there is a SQL standard for `isNumeric` but some SQL servers such as MS and Transact do support `isNumeric(expression)` where expression can actually be a Numeric type (INT, BIGINT, DOUBLE, etc). Maybe @twalthr have more context ?

Reply:
Here I think of it as a string type scalar function, the value represented by this string can be int/bigInt/double and so on.